File: tests/TryExceptOnEmpty.cs
using System;

namespace Demo {
    public partial class TryExceptOnEmpty {
        public static void DoStuff() {
            try
            {
                Console.WriteLine("A");
            }
            catch (Exception E)
            {}
        }
        public static void DoNothingNoSemi() {
            try
            {
                Console.WriteLine("B");
            }
            catch (Exception E)
            {}
        }
    }
}

File: tests/TryExceptOnEmpty.pas
namespace Demo;

interface

uses System;

type
  TryExceptOnEmpty = public class
  public
    class method DoStuff();
    class method DoNothingNoSemi();
  end;

implementation

class method TryExceptOnEmpty.DoStuff();
begin
  try
    Console.WriteLine('A');
  except
    on E: Exception do;
  end;
end;

class method TryExceptOnEmpty.DoNothingNoSemi();
begin
  try
    Console.WriteLine('B');
  except
    on E: Exception do
  end;
end;

end.

